tests/manual/manual_unwrap_or.rs

```diff
@@ -0,0 +1,689 @@
+use crate::test_lint_diagnostics;
```

Comment:
Please add a test that includes a `// comment` in the first arm of the `if let`, or `match` expression. I know we were discussed it but I just want to make sure we include such a test even if the expected behaviour isn't the final one